src/pylint_datetime/__init__.py

```diff
@@ -1,4 +1,185 @@
-# License: MIT
-# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+"""Pylint checker ensure correct usage of datetime module, especially around aware/naive objects."""
 
-"""checker to ensure correct datetime usage with aware objects in datetime module."""
+from typing import Any
+
+import astroid  # type: ignore
+import pylint  # type: ignore
+from pylint.checkers import BaseChecker  # type: ignore
+
+# function calls to fromisoformat, strptime and fromisoformat can't be checked - they parse strings
+# if the specified timezone is None, still allowed although they produce naive objects
+
+
+class DatetimeChecker(BaseChecker):  # type: ignore
+    """class for custom pylint checker."""
+
+    name = "pylint-datetime"
+    priority = -1
+    msgs = {
+        "W9999": (
+            "timedelta() called without keyword arguments",
+            "timedelta-no-keyword-args",
+            "timedelta() called without keyword arguments.",
+        ),
+        "W9998": (
+            'Function call to "%s"/"replace" should be called with a timezone argument',
+            "timezone-no-argument",
+            'Function call to "%s"/"replace" should be called with a timezone argument',
+        ),
+        "W9997": (
+            'Function call to "%s" can only produce naive objects',
+            "naive-datetime-call",
```

Comment:
Maybe `datetime-naive-object-used`?

I wonder if to give better error messages we shouldn't split this into different messages, so we can tell the user which alternative should be used in each case, because as it is now, you just get "don't do this" but you don't get how to do it instead.

This might also be a separate issue and do it as a future improvement.